### PR TITLE
Pin version of MariaDB for DB migrations

### PR DESCRIPTION
**Pull request by sirhcel**
When running database migrations with the latest 10.4.31, they fail as shown in issue #1212. Pinning the point release of MariaDB to the latest known-good version maneuvers around this issue.

### CODE REVIEW

**Comment by juherr** [comment]
This change doesn't fix the issue, it is just hiding it :/

**Comment by sirhcel** [comment]
Yes, you are absolutely right. Let me rephrase the title. This PR aims at shipping a configuration that just works and might save others the time for troubleshooting.

I would leave tackling the actual issues with the migrations and MariaDB 10.4.31 for others.

**Comment by juherr** [comment]
Ok. It makes sense. But why not trying to fix the root issue instead?

At least, add a todo comment that explains it is a workaround with a link to the related issue issue.

**Comment by slachiewicz** [comment]
MariaDB 10.4.31 Docker image was released 4 days ago and it will break environments for everyone pulling fresh 10.4 tags So it's a rational decision to pin to 10.4.30 for the time being.  

Release notes:
https://mariadb.com/kb/en/mariadb-10-4-31-release-notes/
https://mariadb.com/kb/en/mariadb-10-4-31-changelog/

**Comment by slachiewicz** [comment]
MariaDB 10.4.31 Docker image was released 4 days ago and it will break environments for everyone pulling fresh 10.4 tags So it's a rational decision to pin to 10.4.30 for the time being.  

Release notes:
https://mariadb.com/kb/en/mariadb-10-4-31-release-notes/
https://mariadb.com/kb/en/mariadb-10-4-31-changelog/

**Comment by goekay** [comment]
thanks everyone. i agree with the comments here: the PR is not tackling the root cause, but helps to continue providing a working env for builds. this is good enough.

**Comment by goekay** [comment]
obviously, editing docker-compose is not the whole story. our actions are failing due to the same issue. pls see my branch https://github.com/steve-community/steve/tree/experiment_mariadb and the pinned versions in github workflow. i would be happy if you include them in your PR as well.

**Review by slachiewicz — APPROVED** [review]

**Comment by juherr** [comment]
If the issue comes from the migration the only solution will be to remove them.

The v1.0 baseline from #1140 can be used to check it.
It will definitively remove the need of extra rights at the same time.

The huge problem is that it will break all existing instances.
But that's just another argument for multiple flavour releases.

@goekay wdyt?

**Comment by juherr** [comment]
@sirhcel thanks for the todo comment! 🙏
